crates/weaver_forge/templates/test/resource.md

```diff
@@ -39,7 +39,7 @@ Brief: {{ resource.brief }}
 - Sampling relevant: {{ attribute.sampling_relevant }}
   {%- endif %}
   {%- if attribute.deprecated %}
-- Deprecated: {{ attribute.deprecated }}
+- Deprecated: {{ attribute.deprecated.note }}
```

Comment:
Is/will this be a breaking change?

Reply:
Yes, it is a breaking change at the rendering level because, without the `.note`, Jinja renders the new JSON of the `deprecated` field instead of the text that now corresponds to the `note`. I couldn’t find another way to handle this at the rendering level. However, at the semconv parsing level, it is not a breaking change since both formats remain supported.

Reply:
Related comment: https://github.com/open-telemetry/weaver/pull/400/files#r1868713601 ( I thought we'd only keep the `note` on the attribute without introducing one on `deprecated`).

But if we do have `deprecated.note`, it'd still optional thing that does not need to describe the `action`.

I think codegen would need to generate something like 

```java
/*
 * ...
 * @deprecated renamed to {@link
 *     io.opentelemetry.semconv.FooAttributes#FOO_BAR}.
 *  Here's an additional note.
 */
```

from 

```yaml
deprecated: 
  action: renamed
  new_name: foo.bar
note: Here's an additional note.
```

so maybe we should eventually provide some helper filters to render those.

In the scope of this PR, if someone uses `{{ attr.deprecated }}`, I wonder if jinja can call `to_string` on the original `deprecated` object to return a custom string? We'd format it similarly to "Renamed to `foo.bar`", but someone who wants to use new structured info would be able to obtain it with `attr.deprecated.action`, etc

WDYT?